src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/CoverageModelEMWorkspaceGermline.java

```diff
@@ -0,0 +1,102 @@
+
+    /**
+     * Public constructor
+     *
+     * @param rawReadCounts an instance of {@link ReadCountCollection} containing raw read counts
+     * @param germlinePloidyAnnotatedTargetCollection an instance of {@link GermlinePloidyAnnotatedTargetCollection}
+     *                                                for obtaining target ploidies for different sex genotypes
+     * @param sexGenotypeDataCollection an instance of {@link SexGenotypeDataCollection} for obtaining sample sex genotypes
+     * @param params EM algorithm parameters
+     * @param ctx the Spark context
+     * @param copyRatioExpectationsCalculator an implementation of {@link CopyRatioExpectationsCalculator}
+     */
+    public CoverageModelEMWorkspaceGermline(@Nonnull final ReadCountCollection rawReadCounts,
+                                            @Nonnull final GermlinePloidyAnnotatedTargetCollection germlinePloidyAnnotatedTargetCollection,
+                                            @Nonnull final SexGenotypeDataCollection sexGenotypeDataCollection,
+                                            @Nonnull final IntegerCopyNumberExpectationsCalculator copyRatioExpectationsCalculator,
+                                            @Nonnull final CoverageModelEMParams params,
+                                            @Nullable final CoverageModelParameters model,
+                                            @Nullable final JavaSparkContext ctx) {
+        super(rawReadCounts, germlinePloidyAnnotatedTargetCollection, sexGenotypeDataCollection,
+                copyRatioExpectationsCalculator, params, model, ctx);
+    }
+
+    /**
+     * Save copy ratio (here, integer copy number) posteriors to disk
+     *
+     * @param outputPath the output path
+     */
+    protected void saveCopyRatioPosteriors(final String outputPath) {
+        /* fetch forward-backward and Viterbi results */
+        final List<CopyRatioHiddenMarkovModelResults<CoverageModelCopyRatioEmissionData,
+                        IntegerCopyNumberState>> copyRatioHMMResult = getCopyRatioHiddenMarkovModelResults();
+
+        /* perform segmentation */
+        final BiFunction<SexGenotypeData, Target, IntegerCopyNumberState> referenceStateFactory = (sexGenotypeData, target) ->
+                new IntegerCopyNumberState(germlinePloidyAnnotatedTargetCollection.getTargetGermlinePloidyByGenotype(target,
+                        sexGenotypeData.getSexGenotype()));
+        final List<TargetCollection<Target>> sampleTargetCollection = Collections.nCopies(numSamples,
+                new HashedListTargetCollection<>(processedTargetList));
+        final HiddenMarkovModelSegmentProcessor<CoverageModelCopyRatioEmissionData, IntegerCopyNumberState, Target>
+                copyRatioSegmenter = new HiddenMarkovModelSegmentProcessor<>(processedSampleNameList,
+                processedSampleSexGenotypeData, referenceStateFactory, sampleTargetCollection,
+                copyRatioHMMResult.stream()
```

Comment:
Actually, I could just pass a list of `CopyRatioHMMResult` now (`CopyRatioCallingMetadata` contains everything sample-specific).